File: Android/RTMSyncManager/src/main/java/io/agora/rtmsyncmanager/model/AUICustomMsgType.java
package io.agora.rtmsyncmanager.model;

import android.text.TextUtils;

public enum AUICustomMsgType {
    /**
     * 系统消息 成员加入
     */
    AUIChatRoomJoinedMember("AUIChatRoomJoinedMember"),

    ;

    private String name;

    AUICustomMsgType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static AUICustomMsgType fromName(String name) {
        for (AUICustomMsgType type : AUICustomMsgType.values()) {
            if(TextUtils.equals(type.getName(), name)) {
                return type;
            }
        }
        return null;
    }
}

File: Android/IMKitManager/src/main/java/io/agora/imkitmanager/model/AUICustomMsgType.kt
package io.agora.imkitmanager.model

import android.text.TextUtils

enum class AUICustomMsgType(value: String) {
    /**
     * system message: user join chat room
     */
    AUIChatRoomJoinedMember("AUIChatRoomJoinedMember");

    companion object {
        fun fromName(name: String?): AUICustomMsgType? {
            for (type in values()) {
                if (TextUtils.equals(type.name, name)) {
                    return type
                }
            }
            return null
        }
    }
}
